cores/esp32/Print.cpp

```diff
@@ -66,12 +66,11 @@ size_t Print::printf(const char *format, ...)
         }
         len = vsnprintf(temp, len+1, format, arg);
     }
-    write((uint8_t*)temp, len);
     va_end(arg);
-    if(len >= sizeof(loc_buf)) {
+    if(temp != loc_buf){
         free(temp);
     }
-    return len;
+    return write((uint8_t*)temp, len);
```

Comment:
don't do it this way, the buffer has already been freed, `free()` will overwrite the content of `temp` with the delete chain links.

Reply:
oops... absolutely right, a sec

Reply:
you could:
```c++
if(temp != loc_buf){
     len = write((uint8_t*)temp,len);  
     free(temp);
}
return len;
 ```

Reply:
I also want to write(temp) if temp == loc_buf